zebra-scan/src/storage.rs

```diff
@@ -0,0 +1,50 @@
+//! Store viewing keys and results of the scan.
+
+use std::collections::HashMap;
+
+use zebra_chain::{block::Height, transaction::Hash};
+
+/// Store key info and results of the scan.
+#[allow(dead_code)]
+pub struct Storage {
+    /// The key and an optional birthday for it.
+    keys: HashMap<String, Option<Height>>,
+
+    /// The key and the related transaction id.
+    results: HashMap<String, Vec<Hash>>,
```

Comment:
**New ticket:** Do we want to store the decrypted results of scanning in this PR, or do that in a later PR?

These types seem pretty complicated for a first PR, so I will split this into another ticket. It would also be helpful to add type aliases for `SaplingSpend` and `SaplingOutput`, so we can change them later.

```suggestion
    /// The new outputs created in each transaction, for each scanning key. 
    sapling_spend_results: HashMap<String, HashMap<Hash, WalletSaplingSpend>>,

    /// The spent outputs by each transaction, for each scanning key.
    /// If the scanning key is an individual viewing key (without a nullifier deriving key), there are no nullifiers in those results.
    sapling_spend_results: HashMap<String, HashMap<Hash, WalletSaplingOutput<Option<Nullifier>>>>,
```

Reply:
This is now ticket #7944.